library/pkparse.c

```diff
@@ -1239,11 +1246,15 @@ static int pk_parse_key_sec1_der(mbedtls_pk_context *pk,
                                         0)) == 0) {
             if ((ret = pk_get_ecparams(&p, p + len, &params)) != 0 ||
                 (ret = pk_use_ecparams(&params, pk)) != 0) {
+#if !defined(MBEDTLS_PK_USE_PSA_EC_DATA)
```

Comment:
The repeated `#if`s are a bit annoying. Perhaps we should change the structure to `goto exit;` instead of `return`, so that the guarded call to `free()` is only in one place?

Taking a step back, why do we call `ecp_keypair_free()`? Couldn't we call `mbedtls_pk_free()` instead?